project-koku/koku
# [COST-3213] rate limit tasks

## Jira Ticket

[COST-3213](https://issues.redhat.com/browse/COST-3213)

## Description

This change will increase the worker cache timeout and reduce the number of concurrent tasks of a specific type that can be run for a schema for unleash flagged "large customers".

## Testing

0. Set log level to DEBUG
1. Add org1234567 to unleash large-customer at http://localhost:4242/#/features/strategies/cost-management.backend.large-customer
2. Add 
```
        worker_cache.lock_single_task(task_name, [schema_name, provider, "1"], timeout=timeout)
        worker_cache.lock_single_task(task_name, [schema_name, provider, "2"], timeout=timeout)

```
here https://github.com/project-koku/koku/blob/ba3c990cf8885b627e8725238cb373fef2aa75f9/koku/masu/processor/tasks.py#L413
3. Restart worker
Hit report_data masu API for real source and see the task get requeued for being rate limited

## Notes

...
